parser_library/src/context/ordinary_assembly/dependency_collector.cpp

```diff
@@ -121,14 +121,16 @@ dependency_collector& dependency_collector::div_mul(const dependency_collector&
 
 void dependency_collector::adjust_address(address& addr)
 {
-	auto unknown_space = std::find_if(addr.spaces.begin(), addr.spaces.end(), [] (auto& entry) { return entry.first->kind == context::space_kind::LOCTR_UNKNOWN; });
+	auto unknown_space = std::find_if(addr.spaces.begin(), addr.spaces.end(), [](auto& entry) { return entry.first->kind == context::space_kind::LOCTR_UNKNOWN; });
```

Comment:
I suspect that the erase inside the loop causes O(N^2) behavior.
```suggestion
	auto known_spaces = std::partition(addr.spaces.begin(), addr.spaces.end(), [](auto& entry) { return entry.first->kind == context::space_kind::LOCTR_UNKNOWN; });
	std::for_each(known_spaces, addr.spaces.end(), [&addr](auto& entry) { return entry.first->remove_listener(&addr); });
	addr.spaces.erase(known_spaces, addr.spaces.end());
```